### PR TITLE
make SSH logging in e2e more consistent

### PR DESCRIPTION
**Pull request by thockin**

### CODE REVIEW

File: test/e2e/reboot.go [hunk 1]
```diff
@@ -162,8 +162,9 @@ func issueSSHCommand(node *api.Node, provider, cmd string) error {
 		return fmt.Errorf("couldn't find external IP address for node %s", node.Name)
 	}
 	Logf("Calling %s on %s", cmd, node.Name)
-	if _, _, code, err := SSH(cmd, host, provider); code != 0 || err != nil {
-		return fmt.Errorf("when running %s on %s, got %d and %v", cmd, node.Name, code, err)
+	if result, err := SSH(cmd, host, provider); result.Code != 0 || err != nil {
+		LogSSHResult(result)
+		return err
```
Comment (ixdy) [inline test/e2e/reboot.go]: if err is `nil` but `result.Code` is nonzero, this will appear like a success.

File: test/e2e/resize_nodes.go [hunk 2]
```diff
@@ -338,14 +338,13 @@ func performTemporaryNetworkFailure(c *client.Client, ns, rcName string, replica
...
+			result, err := SSH(undropCmd, host, testContext.Provider)
+			if result.Code == 0 && err == nil {
 				return true, nil
-			} else {
-				Logf("Expected 0 exit code and nil error when running '%s' on %s, got %d and %v",
-					undropCmd, node.Name, code, err)
-				return false, nil
 			}
+			LogSSHResult(result)
+			Logf("Unexpected error: %v", err)
```
Comment (ixdy) [inline test/e2e/resize_nodes.go]: again here: this may say "Unexpected error: nil" on failures.

File: test/e2e/reboot.go (hunk 1 above)
Comment (thockin) [inline test/e2e/reboot.go]: good catch

**Comment by ixdy** [comment]
Since almost every one of these instances checks to see if `result.Code` is 0, I wonder if `SSH()` should do this itself, logging the output and raising an error if the code doesn't match this. (Or probably need 2 functions, since occasionally we expect something besides 0, but the default should be 0.)

File: test/e2e/resize_nodes.go (hunk 2 above)
Comment (thockin) [inline test/e2e/resize_nodes.go]: fixed for next push

**Comment by thockin** [comment]
I though about that.  This certainly leaves something to be desired, I was just trying to make as small a change as I could to make debug better.  Our e2e utils overall are pretty inconsistent.  Plumbing an expected code or a test function seemed like too much churn, but I'll take a look.

File: test/e2e/reboot.go [hunk 3]
```diff
@@ -162,8 +162,9 @@ func issueSSHCommand(node *api.Node, provider, cmd string) error {
 		return fmt.Errorf("couldn't find external IP address for node %s", node.Name)
 	}
 	Logf("Calling %s on %s", cmd, node.Name)
-	if _, _, code, err := SSH(cmd, host, provider); code != 0 || err != nil {
-		return fmt.Errorf("when running %s on %s, got %d and %v", cmd, node.Name, code, err)
+	if result, err := SSH(cmd, host, provider); result.Code != 0 || err != nil {
+		LogSSHResult(result)
+		return fmt.Errorf("failed running %q: %v (exit code %d)", cmd, err, code)
```
Comment (ixdy) [inline test/e2e/reboot.go]: result.Code

**Comment by thockin** [comment]
rebased

**Comment by thockin** [comment]
I don't have bandwidth for further polishing of this.  If you think this is useful, LGTM it.  If you'd rather wait for a better patch, feel free to close it.

**Comment by ixdy** [comment]
LGTM. there's a few places where we may still log "unexpected error: nil" but this is still better than the status quo.

**Comment by thockin** [comment]
I didn't mean to sound snarky - I just have too much to do :)

**Pull request merged**